diagrams/request_uri_mode_create.md

```diff
@@ -0,0 +1,49 @@
+```plantuml
+@startuml
+
+autonumber
+
+box "Wallet"
+participant "Metadata" as wm
+participant "Authorization Endpoint" as w
+end box
+
+participant "User Agent" as u
+
+box "Verifier"
+participant "Frontend" as r
+participant "Request Endpoint" as rp
+participant "Response Endpoint" as rb
+end box
+
+u --> r : use
+activate r
+
+r --> u: **signed authorization request**\n(client_id, request_uri, request_uri_mode=create, state)
+deactivate r
+u --> w: **signed authorization request**\n(client_id, request_uri, request_uri_mode=create, state)
+activate w
+w --> w: check authorization request signature
+w --> w: check on trustworthiness of Verifier (approach on trust mechanism)
+w --> rp: POST **request_uri** (\n[OPTIONALstate,\n[OPTIONAL]issuer, \n[OPTIONAL]wallet_metadata, \n[OPTIONAL]w_nonce, \n[OPTIONAL]w_ephm_key)
+rp --> wm: [OPTIONAL] get wallet metadata
+wm --> rp: [OPTIONAL] wallet metadata
+rp -> rp: create and sign (and optionally encrypt) presentation request object (client_id, w_nonce, nonce, response_uri, \npresentation_definition, state)
+rp --> w: **signed (optionally encrypted) request object** (client_id, w_nonce, nonce, response_uri, \npresentation_definition, state)
+note over u, w: do we want to allow unsigned presentation request objects, too?
+w -> w: authenticate and\n authorize Verifier
+
+note over u, w: user authentication and credential selection/confirmation
+
+w -> w: create verifiable\npresentation (credential)
+w --> rb: POST response \n(vp_token, presentation_submission, state)
+rb --> w: redirect_url
+w --> u: response (response_code)
+u --> r: response (response_code)
+activate r
+r --> rb: get presentation response\n (transaction_id, response_code)
+rb --> r: presentation response
+r -> r: validate presentation \n(incl. nonce binding)
+r -> r: use presented credential 
+@enduml
```

Comment:
In general, I like the `request_uri_mode` approach much better but here are a few comments on the diagram:

1. Not clear why request in step 2 has to be signed. This does not add any security. I'd rather provide the Authz Request parameters unsigned. NOTE: JAR does not sign `client_id` either if `request_uri` (or `request`) is used in the Authorization Request.
2. (Can be ignored if signing gets removed, see 1.) What object is used to convey the signed Authz Request in step 2? Is it `request` as defined in JAR? NOTE: For OAuth2 backwards compatibility, `client_id` has to be provided always as a query string parameter (outside of the Request Object) as well.
3. We need to add similar language as described in Connect/JAR that says that if the `request_uri` Request Object contains the same parameters as in the initial Authz Request, the `request_uri` Request Object has priority.
4. I believe there is a typo in step 2: `request_uri_mode=create`, it should rather be `request_uri_mode=POST`.
5. I would rename the `request_uri_mode=POST` to `request_uri_mode=post_metadata_exchange`. This indicates additional AS metadata is provided plus a `w_nonce`.
6. Remove `issuer` in step 6 since the `issuer` is REQUIRED in `wallet_metadata`.
7. Remove `w_ephm_key` from step 6 since `jwks`/`jwks_uri` can be included in `wallet_metadata`. The wallet acts as an AS and makes their keys discoverable.
6. Rename `w_nonce` to `as_nonce`.
7. To your question between step 10 and 11, imo, we should allow unsigned Authz Request objects but then the same rules as in JAR apply which is setting a dedicated parameter `require_signed_request_object` to `false` in the Authz Request.
8. In step 13, it might be good to show that `nonce` is included in the `vp_token`.
9. In step 15, I'd remove the `response_code` until made it REQUIRED or normatively defined. IMO, this is required before we do ID-3.
10. Between step 13 and step 14, something is missing.
11. Change "response" in step 15 to "redirect"; remove `response_code` for the same reason as given in 9.
12. Same as above but applied to step 16.
13. Remove steps 17-20 since they are out-of-scope of OID4VP atm.